### PR TITLE
test(header-cell): add more testing to header-cell

### PR DESCRIPTION
**Pull request by alexw10**
**What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ X ] Other... Please describe:
Add more tests to improve the quality.

**What is the current behavior?** (You can also link to an open issue here)
Testing coverage is rather low in header-cell around ~60%

**What is the new behavior?**
Testing coverage is now around 95% in header-cell

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [ X ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
I work at a company and my team is currently pulling NGX Datatable into our component library and making it our own. Basically figured it would be a good starting point for us. We are high on testing where we are and the first thing we are doing is going through and improving the testing being done on the components.

This is the first PR of many we plan to do. Probably throughout your whole project.
This is basically the first test run to see if you guys will welcome this kind of change/update to your tests.

Please let us know as soon as you can, if you don't want these improvements for some reason we will simply just not do PR's to you guys for these updates.

Here is the old and new test coverage output after my updates:
OLD:
Statements   : 65.13% ( 1218/1870 )
Branches     : 35.88% ( 254/708 )
Functions    : 66.67% ( 220/330 )
Lines        : 66.71% ( 1116/1673 )
  
NEW:
Statements   : 66.83% ( 1249/1869 )
Branches     : 39.86% ( 283/710 )
Functions    : 67.78% ( 223/329 )
Lines        : 68.44% ( 1147/1676 )

-- Your Friends At Veritas 😄

### CODE REVIEW

File: src/components/header/header-cell.component.ts [hunk 1]
```diff
@@ -1,54 +1,48 @@
...
+                class="datatable-header-cell-label draggable"
+                (click)="onSort()"
+                [innerHTML]="name">
         </span>
       </span>
-      <ng-template
-        *ngIf="column.headerTemplate"
-        [ngTemplateOutlet]="column.headerTemplate"
-        [ngOutletContext]="{ 
+          <ng-template
```
Comment (amcdnl) [inline src/components/header/header-cell.component.ts]: Can you fix these indents to be 2 tabs please?

Comment (alexw10) [inline src/components/header/header-cell.component.ts]: Yup, no problem at all, will update. You want it to be 2 tabs everywhere?

Comment (alexw10) [inline src/components/header/header-cell.component.ts]: Yup, no problem at all, will update. You want it to be 2 tabs everywhere?

Comment (alexw10) [inline src/components/header/header-cell.component.ts]: Updated, let me know if any other changes are needed.

File: src/components/header/header-cell.component.ts [hunk 2]
```diff
@@ -1,9 +1,7 @@
-import {
-  Component, Input, EventEmitter, Output, HostBinding, HostListener
```
Comment (arlowhite) [inline src/components/header/header-cell.component.ts]: @amcdnl This just broke the build. `HostListener` import is needed.

This PR is reverting some of the earlier header context menu code.

**Comment by amcdnl** [comment]
I have reverted this PR per the items described by @arlowhite .

@alexw10 - can you make a new PR and implement the items he has commented on.

**Comment by alexw10** [comment]
Hi @amcdnl, sorry I didn't get notified of this, it says merged, so I am wondering if this is merged in or should I create a new PR still to fix the issues?

**Comment by alexw10** [comment]
Ah, nvm just looked. Looks like it was reverted.
I will put up a PR soon hopefully with the fixes @arlowhite mentioned.

I think npm run package should really run as part of your build or checks as well then shouldn't it? just so things like this get caught?

**Pull request merged**